amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -203,6 +209,31 @@ private void updateLeasesForChildShards()
         log.info("Shard {}: Updated current lease {} with child shard information: {}", shardInfo.shardId(), currentLease.leaseKey(), childShardIds);
     }
 
+    private void garbageCollectLeaseForParentShard() throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+       final Lease parentShardLease = leaseCoordinator.getCurrentlyHeldLease(leaseKeyProvider.apply(shardInfo));
+
+        if (parentShardLease.childShardIds().isEmpty()) {
+            log.warn("No child shard state found in lease {} - not deleting lease.", parentShardLease);
+        } else {
+            final Set<String> expectedLeaseKeys = parentShardLease.childShardIds().stream()
+                    .map(c -> ShardInfo.getLeaseKey(shardInfo, c)).collect(Collectors.toSet());
+            final Set<String> leaseKeys = new HashSet<>();
+
+            for (String childShard : parentShardLease.childShardIds()) {
+                final String leaseKey = ShardInfo.getLeaseKey(shardInfo, childShard);
+                leaseKeys.add(leaseCoordinator.leaseRefresher().getLease(leaseKey).leaseKey());
+            }
+
+            if (Objects.equals(expectedLeaseKeys, leaseKeys)) {
```

Comment:
Does Objects.equals look at value or reference? If reference, I think this comparison would always fail since we're regenerating the leaseKeys in expected and actuals

Reply:
Nevermind, from unit tests I'm assuming this is doing the right thing

Reply:
Checks both; if either the references are the same or the `.equals()` are the same.